### ISSUE TITLE
Cannot build mesh as INLA is not installed. 

### ISSUE THREAD

**Issue by helhafyani** [aknandi/disaggregation#47]
Hello,

I am following your example on leukemia incidence data, and when I got to the prepare_data function step and execute the function, I got this error:

Cannot build mesh as INLA is not installed. If you need a spatial field in your model, you must install INLA.

I have installed the "INLA" package and called it before executing the function but I got the same error. I have restarted R and re-run everything but the error remains.

Any idea why?

Many thanks!
Hafsa

**Comment by timcdlucas** [comment]
Hmmmm sounds odd.

Could I please confirm that you installed INLA using the commands here:
http://www.r-inla.org/download

and that if you do 
library(INLA)
sessionInfo()

you can see INLA as a loaded package? If you just did install.packages('INLA') it won't have installed as it isn't on CRAN.

The code isn't doing anything particularly clever at that point and should just load INLA if it is installed. Otherwise I'll have to think more carefully.

**Comment by helhafyani** [comment]
Thank you for your response.

I did install the stable version of INLA from the R-repository as shown in the link.

After runing the commands, I got:

>library(INLA)
Error: package or namespace load failed for ‘INLA’ in rbind(info, getNamespaceInfo(env, "S3methods")):
 number of columns of matrices must match (see arg 2)
In addition: Warning message:
le package ‘INLA’ a été compilé avec la version R 3.6.0 

> sessionInfo()
R version 3.5.2 (2018-12-20)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=French_France.1252  LC_CTYPE=French_France.1252    LC_MONETARY=French_France.1252
[4] LC_NUMERIC=C                   LC_TIME=French_France.1252    

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] Matrix_1.2-18        raster_3.0-7         dplyr_0.8.3          SpatialEpi_1.2.3    
[5] sp_1.3-2             disaggregation_0.1.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.3         lattice_0.20-38    deldir_0.1-23      class_7.3-15      
 [5] gtools_3.8.1       assertthat_0.2.1   digest_0.6.23      foreach_1.4.7     
 [9] R6_2.4.1           coda_0.19-3        e1071_1.7-3        ggplot2_3.2.1     
[13] pillar_1.4.3       rlang_0.4.2        lazyeval_0.2.2     spdep_1.1-3       
[17] rstudioapi_0.10    gdata_2.18.0       gmodels_2.18.1     labeling_0.3      
[21] splines_3.5.2      rgdal_1.4-8        foreign_0.8-74     munsell_0.5.0     
[25] compiler_3.5.2     pkgconfig_2.0.3    rgeos_0.5-2        tidyselect_0.2.5  
[29] tibble_2.1.3       expm_0.999-4       codetools_0.2-16   viridisLite_0.3.0 
[33] crayon_1.3.4       sf_0.8-0           MASS_7.3-51.5      grid_3.5.2        
[37] nlme_3.1-137       spData_0.3.2       gtable_0.3.0       lifecycle_0.1.0   
[41] DBI_1.1.0          magrittr_1.5       units_0.6-5        scales_1.1.0      
[45] KernSmooth_2.23-16 farver_2.0.3       LearnBayes_2.15.1  doParallel_1.0.15 
[49] boot_1.3-24        iterators_1.0.12   tools_3.5.2        glue_1.3.1        
[53] purrr_0.3.3        yaml_2.2.0         colorspace_1.4-1   maptools_0.9-9    
[57] classInt_0.4-2

**Comment by timcdlucas** [comment]
OK, for some reason installing INLA didn't work for you.

Googling the error message I found this page:
https://github.com/tidyverse/tidyverse/issues/179

Maybe you have packages installed from 3.5 and 3.6 or something.

https://r.789695.n4.nabble.com/Error-in-rbind-info-getNamespaceInfo-env-quot-S3methods-quot-td4755490.html
This thread has a fairly reaosonable looking suggestion for working out which package is causing issues.

Basically you need to get to the point where:
library(INLA)
?inla

runs properly. Then I think disaggregation should work.

I hope that helps.

**Comment by helhafyani** [comment]
Thank you for response.

I will check those links and I hope I could fix the problem.

Many thanks!

**Comment by timcdlucas** [comment]
Hopefully this is working for you now so I will close the issue.

Thanks for getting in touch!

**Comment by helhafyani** [comment]
Thank you very much for your response.
